salt/runner.py

```diff
@@ -301,9 +301,10 @@ def _run_runner(self):
             if not self.opts.get('quiet', False):
                 display_output(ret, 'nested', self.opts)
         else:
-            ret = {
-                'retcode': salt.defaults.exitcodes.EX_SOFTWARE,
-            }
+            if not ret:
```

Comment:
> fwiw this change is unrelated to PR, I just made it to fix the failing tests

Ah. Well, took a look at the code and maybe this will be a good improvement (and 🤞 help future edits not introduce bugs :trollface:)

```suggestion
            # If we don't have any values in ret by now, that's a problem.
            # Otherwise, we shouldn't be overwriting the retcode.
            if not ret:
```

Reply:
added comment, thanks!